### PR TITLE
SmtpClientTest.TestZeroTimeout: FailFast to get CI coredump

### PR DESCRIPTION
**Pull request by tmds**
To help debug https://github.com/dotnet/corefx/issues/40711

CC @ViktorHofer @stephentoub @wfurt

### CODE REVIEW

File: src/System.Net.Mail/tests/Functional/SmtpClientTest.cs [hunk 1]
```diff
@@ -317,21 +317,33 @@ public void TestMailDelivery()
...
+            });
+            // Abort in order to get a coredump if this test takes too long.
+            if (!testTask.Wait(TimeSpan.FromMinutes(5)))
+            {
+                Environment.FailFast(nameof(TestZeroTimeout));
+            }
+            else
+            {
+                await testTask;
             }
```
Comment (stephentoub) [inline src/System.Net.Mail/tests/Functional/SmtpClientTest.cs]: What's the purpose of this else block?

Comment (tmds) [reply src/System.Net.Mail/tests/Functional/SmtpClientTest.cs]: I'll remove it. I had misassumed `.Wait` wouldn't throw in the completed case.

**Comment by ViktorHofer** [comment]
Can we move the test to Outerloop so that the crash doesn't interfere with our CI?

**Comment by tmds** [comment]
> Can we move the test to Outerloop so that the crash doesn't interfere with our CI?

I don't think it reproduces often, I tried 1000 times. Let's run it a couple of times here before merging.

**Comment by ViktorHofer** [comment]
It reprod the day after the original change was done. Trying to force the hang here in the PR sounds good.

File: src/System.Net.Mail/tests/Functional/SmtpClientTest.cs [hunk 2]
```diff
@@ -317,21 +317,29 @@ public void TestMailDelivery()
...
 
         [Fact]
-        [ActiveIssue(40711)]
+        // [ActiveIssue(40711)]
         [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, ".NET Framework has a bug and may not time out for low values")]
         [PlatformSpecific(~TestPlatforms.OSX)] // on OSX, not all synchronous operations (e.g. connect) can be aborted by closing the socket.
         public void TestZeroTimeout()
         {
-            using (Socket serverSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            var testTask = Task.Run(() =>
```
Comment (davidsh) [inline src/System.Net.Mail/tests/Functional/SmtpClientTest.cs]: nit: 'var' shouldn't be used here. It's only used when the type is obvious (from a `new` or a cast).

**Comment by tmds** [comment]
Unrelated fails on netcoreapp-Windows_NT-Debug-x64-(Windows.Nano.1809.Amd64.Open)windows.10.amd64.serverrs5.open

- System.Security.Cryptography.Rsa.Tests.EncryptDecrypt_Array.RsaCryptRoundtrip_OaepSHA384
```
   at System.Security.Cryptography.RSAImplementation.RSACng.EncryptOrDecrypt(SafeNCryptKeyHandle key, ReadOnlySpan`1 input, AsymmetricPaddingMode paddingMode, Void* paddingInfo, Boolean encrypt) in /_/src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs:line 245
   at System.Security.Cryptography.RSAImplementation.RSACng.EncryptOrDecrypt(Byte[] data, RSAEncryptionPadding padding, Boolean encrypt) in /_/src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs:line 119
   at System.Security.Cryptography.RSAImplementation.RSACng.Decrypt(Byte[] data, RSAEncryptionPadding padding) in /_/src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs:line 30
   at System.Security.Cryptography.Rsa.Tests.EncryptDecrypt_Array.Decrypt(RSA rsa, Byte[] data, RSAEncryptionPadding padding) in /_/src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.cs:line 15
   at System.Security.Cryptography.Rsa.Tests.EncryptDecrypt.RsaCryptRoundtrip(RSAEncryptionPadding paddingMode, Boolean expectSuccess) in /_/src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.cs:line 332
   at System.Security.Cryptography.Rsa.Tests.EncryptDecrypt.RsaCryptRoundtrip_OaepSHA384() in /_/src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.cs:line 310
```

- System.Security.Cryptography.Algorithms.Tests Work Item

```
no stack trace
```

**Comment by tmds** [comment]
@ViktorHofer can you please give this a few runs? I don't have the privileges.

**Comment by davidsh** [comment]
Outerloop test failures are due to #41108.

**Comment by tmds** [comment]
unrelated CI failures.
On Windows, same as before.
On Fedora:
- System.Net.WebSockets.Client.Tests.CancelTest.CloseOutputAsync_Cancel_Success(server: ws://corefx-net-http11.azurewebsites.net/WebSocket/EchoWebSocket.ashx)
Failed

```
Assert.Equal() Failure\nExpected: Aborted\nActual:   Open
   at System.Net.WebSockets.Client.Tests.ClientWebSocketTestBase.TestCancellation(Func`2 action, Uri server) in /_/src/System.Net.WebSockets.Client/tests/ClientWebSocketTestBase.cs:line 76
   at System.Net.WebSockets.Client.Tests.CancelTest.CloseOutputAsync_Cancel_Success(Uri server) in /_/src/System.Net.WebSockets.Client/tests/CancelTest.cs:line 97
--- End of stack trace from previous location where exception was thrown ---
```
- System.Net.Sockets.Tests.SelectTest.Select_AcceptNonBlocking_Success
```
System.TimeoutException : Task timed out after 00:00:05
   at System.Threading.Tasks.TaskTimeoutExtensions.TimeoutAfter(Task task, TimeSpan timeout) in /_/src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 42
   at System.Net.Sockets.Tests.SelectTest.Select_AcceptNonBlocking_Success() in /_/src/System.Net.Sockets/tests/FunctionalTests/SelectTest.cs:line 268
--- End of stack trace from previous location where exception was thrown ---
```
On mac:
- System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandlerTest.GetAsync_UnicodeHostName_SuccessStatusCodeInResponse
```
System.Net.Http.HttpRequestException : An error occurred while sending the request.\n---- System.IO.IOException : The response ended prematurely.
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs:line 726
   at System.Net.Http.HttpConnectionPool.SendWithNtConnectionAuthAsync(HttpConnection connection, HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs:line 565
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs:line 536
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs:line 33
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts) in /_/src/System.Net.Http/src/System/Net/Http/HttpClient.cs:line 521
   at System.Net.Http.Functional.Tests.HttpClientHandlerTest.GetAsync_UnicodeHostName_SuccessStatusCodeInResponse() in /_/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs:line 1659
--- End of stack trace from previous location where exception was thrown ---
----- Inner Stack Trace -----
   at System.Net.Http.HttpConnection.FillAsync() in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs:line 1445
   at System.Net.Http.HttpConnection.ReadNextResponseHeaderLineAsync(Boolean foldedHeadersAllowed) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs:line 1343
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs:line 513
```
- System.Net.Http.Functional.Tests.PlatformHandler_HttpClientHandlerTest.GetAsync_UnicodeHostName_SuccessStatusCodeInResponse
```
System.Net.Http.HttpRequestException : An error occurred while sending the request.\n---- System.Net.Http.CurlException : Server returned nothing (no headers, no data)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts) in /_/src/System.Net.Http/src/System/Net/Http/HttpClient.cs:line 521
   at System.Net.Http.Functional.Tests.HttpClientHandlerTest.GetAsync_UnicodeHostName_SuccessStatusCodeInResponse() in /_/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs:line 1659
--- End of stack trace from previous location where exception was thrown ---
----- Inner Stack Trace -----
   at System.Net.Http.CurlHandler.ThrowIfCURLEError(CURLcode error) in /_/src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.cs:line 655
   at System.Net.Http.CurlHandler.MultiAgent.FinishRequest(StrongToWeakReference`1 easyWrapper, CURLcode messageResult) in /_/src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.MultiAgent.cs:line 881
```

**Comment by tmds** [comment]
@ViktorHofer you want to give this a few more runs? Or merge it to master?

**Comment by tmds** [comment]
@ViktorHofer do you want to merge this to master?

**Comment by ViktorHofer** [comment]
Yes, I think that would make more sense.

**Comment by ViktorHofer** [comment]
thanks @tmds

**Pull request merged**